risc0/zkvm/sdk/rust/guest/src/env.rs

```diff
@@ -107,28 +107,96 @@ pub fn send_recv_as_u32(channel: u32, buf: &[u8]) -> (&'static [u32], usize) {
     ENV.get().send_recv_as_u32(channel, buf)
 }
 
-/// Read private data from the host.
+/// Reads private data from the host.
+///
+/// # Examples
+/// Values are read in the order in which they are written by the host),
+/// as in a queue. In the following example, `first_value` and `second_value`
+/// have been shared consecutively by the host via `env::write`.
+/// ```rust, ignore
+/// let first_value: u64 = env::read();
+/// let second_value: u64 = env::read();
+/// ```
+/// For ease and clarity, we recommend sharing multiple values between guest and
+/// host as a struct. As a practical example, in our [digital signature example](https://github.com/risc0/risc0-rust-examples/tree/main/digital-signature),
+/// we read in a `SigningRequest` that contains both a `Passphrase` and a
```

Comment:
It's be better if we had a self-contained example so that:
1. it requires less context to understand the idea
2. it doesn't depend on another repository where the link might eventually fail